### ISSUE TITLE
Open CV2 Colmap pose Conversion

### ISSUE THREAD

**Issue by engrmusawarali71** [colmap/colmap#3088]
I am trying to convert the poses from Cv2 to colmap. do the following steps:

def rotmat2qvec(R):
    Rxx, Ryx, Rzx, Rxy, Ryy, Rzy, Rxz, Ryz, Rzz = R.flat
    K = np.array([
        [Rxx - Ryy - Rzz, 0, 0, 0],
        [Ryx + Rxy, Ryy - Rxx - Rzz, 0, 0],
        [Rzx + Rxz, Rzy + Ryz, Rzz - Rxx - Ryy, 0],
        [Ryz - Rzy, Rzx - Rxz, Rxy - Ryx, Rxx + Ryy + Rzz]]) / 3.0
    eigvals, eigvecs = np.linalg.eigh(K)
    qvec = eigvecs[[3, 0, 1, 2], np.argmax(eigvals)]
    if qvec[0] < 0:
        qvec *= -1
    return qvec

def opencv_to_colmap(R_opencv, t_opencv):
  #  R_colmap = R_opencv.T
  #  t_colmap = -R_colmap @ t_opencv
    quat = rotmat2qvec(R_opencv)  # Returns [qw, qx, qy, qz] #
    quat_colmap = [quat[0], quat[1], quat[2], quat[3]]  # COLMAP format is [qw, qx, qy, qz]
    return quat_colmap, t_opencv

I wanted to know, if this is correct way to convert the rotation and translation.

**Comment by engrmusawarali71** [comment]
@tsattler

**Comment by starInUniverse** [comment]
> I am trying to convert the poses from Cv2 to colmap. do the following steps:
> 
> def rotmat2qvec(R): Rxx, Ryx, Rzx, Rxy, Ryy, Rzy, Rxz, Ryz, Rzz = R.flat K = np.array([ [Rxx - Ryy - Rzz, 0, 0, 0], [Ryx + Rxy, Ryy - Rxx - Rzz, 0, 0], [Rzx + Rxz, Rzy + Ryz, Rzz - Rxx - Ryy, 0], [Ryz - Rzy, Rzx - Rxz, Rxy - Ryx, Rxx + Ryy + Rzz]]) / 3.0 eigvals, eigvecs = np.linalg.eigh(K) qvec = eigvecs[[3, 0, 1, 2], np.argmax(eigvals)] if qvec[0] < 0: qvec *= -1 return qvec
> 
> def opencv_to_colmap(R_opencv, t_opencv):
> 
> ```
> quat = rotmat2qvec(R_opencv)  # Returns [qw, qx, qy, qz] #
> quat_colmap = [quat[0], quat[1], quat[2], quat[3]]  # COLMAP format is [qw, qx, qy, qz]
> return quat_colmap, t_opencv
> ```
> 
> I wanted to know, if this is correct way to convert the rotation and translation.
> 
> [@tsattler](https://github.com/tsattler) [@ahojnnes](https://github.com/ahojnnes)

I want to konw about this issue, too. Have you made any progress?

**Comment by engrmusawarali71** [comment]
Yes, I did conversion. It is only you who has to convert rotation matrices to quats.

**Comment by ahojnnes** [comment]
Closing: OpenCV/COLMAP use same coordinate convention; just quaternion conversion needed.
